src/controllers/PostCommentController.js

```diff
@@ -0,0 +1,121 @@
+import { PAGE_LIMIT } from '../constants/shared';
+
+/**
+ * @description PostComment Controller class
+ */
+export default class PostCommentController {
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async createPostComment(req, res) {
+    const { currentUser, body, post } = req;
+    body.author = currentUser._id;
+    body.post = post._id;
+    const postComment = await PostComment.create(body);
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      postComment,
+      message: responseMessages.created('Post Comment'),
+    });
+  }
+
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async updatePostComment(req, res) {
+    const { currentUser, body, postComment } = req;
+
+    if (!currentUser._id.equals(postComment.author)) {
+      return res.status(statusCodes.UNAUTHORIZED).json({
+        status: statusCodes.UNAUTHORIZED,
+        message: responseMessages.unauthorized(),
+      });
+    }
+
+    await postComment.updateOne(body);
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      postComment: {
+        ...postComment.toObject(),
+        ...body,
+      },
+      message: responseMessages.updated('Post Comment'),
+    });
+  }
+
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async deletePostComment(req, res) {
+    const { currentUser, postComment } = req;
+
+    if (!currentUser._id.equals(postComment.author)) {
+      return res.status(statusCodes.UNAUTHORIZED).json({
+        status: statusCodes.UNAUTHORIZED,
+        message: responseMessages.unauthorized(),
+      });
+    }
+
+    await postComment.deleteOne();
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      postComment,
+      message: responseMessages.deleted('Post Comment'),
+    });
+  }
+
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async getAllPostComments(req, res) {
+    const {
+      post: { _id: post },
+    } = req;
+    const { page = 1 } = req.query;
+
+    const comments = await PostComment.paginate(
+      {
+        post,
+        status: {
+          $ne: 'deleted',
+        },
+      },
+      {
+        populate: [{ path: 'author', select: 'username email picture' }],
+        limit: PAGE_LIMIT,
+        offset: page - 1,
+      },
+    );
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      ...comments,
+      postComments: comments.docs,
+      docs: undefined,
+      offset: undefined,
+      limit: undefined,
```

Comment:
@oesukam  Why are you returning these 3 fields as undefined?

Reply:
@gracelungu the pagination plugin returns those fields that are unnecessary to be in the response body